records_mover/utils/concat_files.py

```diff
@@ -18,10 +20,23 @@ def readable(self) -> bool:
         return True
 
     def readall(self) -> bytes:
-        return b''.join(f.read(-1) for f in self._files)
+        out = b''
+        while self._files:
+            f = self._files.pop(0)
+            chunk = f.read()
+            self._tell += len(chunk)
+            out = out + chunk
+        return out
+
+    def tell(self) -> int:
+        return self._tell
 
     def read(self, size: int = -1) -> bytes:
-        if size < 0:
+        #  "When size is omitted or negative, the entire contents of
+        #  the file will be read and returned"
+        #
+        # https://docs.python.org/3/tutorial/inputoutput.html
+        if size == -1:
```

Comment:
Hmm... Not understanding this change. From the language quoted above, seems like this should stay as `size < 0`? I can't imagine it comes up often, what kind of animal passes in size=-2, but specifications 🤷‍♂️ 

```suggestion
        if size < 0:
```